test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/TxnVerbs.java

```diff
@@ -330,8 +355,64 @@ public static HapiContractUpdate contractUpdate(String contract) {
 		return new HapiContractUpdate(contract);
 	}
 
+	public static HapiSpecOperation contractDeploy(final String contractName) {
+		return withOpContext((spec, ctxLog) -> {
+			final var path = String.format(RESOURCE_PATH + ".bin", contractName);
+			validateFileExists(path);
+			final var payer = cryptoCreate("PAYER");
+			final var file = fileCreate(contractName);
+			final var updatedFile = updateLargeFile("PAYER", contractName, extractByteCode(path));
+			final var deployedContract = contractCreate(contractName).bytecode(contractName);
+			allRunFor(spec, payer, file, updatedFile, deployedContract);
+		});
+	}
+
+	public static HapiSpecOperation nestedContractDeploy(final String contractName, Object... params) {
+		return withOpContext((spec, ctxLog) -> {
+			final var path = String.format(RESOURCE_PATH + ".bin", contractName);
+			validateFileExists(path);
+			final var payer = cryptoCreate("PAYER");
+			final var file = fileCreate(contractName);
+			final var updatedFile = updateLargeFile("PAYER", contractName, extractByteCode(path));
+			final var constructorABI = getABIFor(CONSTRUCTOR, EMPTY, contractName);
+			final var deployedContract = contractCreate(contractName, constructorABI, params).bytecode(contractName);
+			allRunFor(spec, payer, file, updatedFile, deployedContract);
+		});
+	}
+
+
 	/* SYSTEM */
 	public static HapiFreeze hapiFreeze(final Instant freezeStartTime) {
 		return new HapiFreeze().startingAt(freezeStartTime);
 	}
+
+	private static String getABIFor(final FunctionType type, final String functionName, String contractName) {
+		final var path = String.format(RESOURCE_PATH + ".json", contractName);
+		validateFileExists(path);
+		var ABI = EMPTY;
+		try (final var input = new FileInputStream(path)) {
+			final var array = new JSONArray(new JSONTokener(input));
+			ABI = IntStream
+					.range(0, array.length())
+					.mapToObj(array::getJSONObject)
+					.filter(object -> type == CONSTRUCTOR
+							? object.getString("type").equals(type.toString().toLowerCase())
+							: object.getString("type").equals(type.toString().toLowerCase()) && object.getString("name").equals(functionName))
+					.map(JSONObject::toString)
+					.findFirst()
+					.orElseThrow(() -> new IllegalArgumentException("No such function found: " + functionName));
+		} catch (IOException e) {
+			e.getStackTrace();
+		}
+		return ABI;
+	}
+
+	private static void validateFileExists(String path) {
+		final var file = new File(path);
+		if (!file.exists()) {
+			throw new IllegalArgumentException("Invalid argument: " + path.substring(path.lastIndexOf('/') + 1));
+		}
+	}
+
+	public enum FunctionType {CONSTRUCTOR, FUNCTION}
```

Comment:
I suggest having different methods `getAbiForFunction` and `getAbiForConstructor`; then we won't need the `FunctionType` enum and we can simplify the logic in `getABIFor`